mcode/mcode-coverage-checker
# Steam 909/semantic filters

# Description
Issue: 909

Implements constraint-based filtering for all mCODE resource types

## (Bugfix) Solution
The functionality still uses the meta profiled filters if they produce valid results, but will fall back on the constraint-based filters if the meta-based filters fail. Each contraint-based filter references the IG where appropriate. Roughly, there are three kinds of filters: 
- A resource has a particular code that identifies it as an mCODE concept;
- A resource has a code from a codesystem from an mCODE ValueSet; 
- A resource references another resource in another way which identifies it as an mCODE concept.

## Important Changes
Please list important files (meaning substantial or integral to the PR) along with a list of the general changes that should be highlighted for reviewers.

- New constraint-based filters for each resource filter 
- Adds valuesets used in filtering, as well as code for expanding future valuesets (borrowed from the MEF)
- Adds a ValueSetChecker class which references imported VS's to check if a given code is in a particular valueset 
- Moves resourceUtils up a folder, ensuring all files in the coverageChecker folder are doing actual coverage checking.
- Renames the resourceUtils files to resourceFilter 

## Testing Recommendations
- Ensure that this new approach doesn't change any of the statistics you can see on the [live demo](https://mcode.github.io/mcode-coverage-checker/)
- Ensure codes used in filtering match the codes mentioned in the IG 
- Pause to consider if there is a more comprehensive filter for a given resource.

# Checklists

**Submitter:**

@ACCT1 :
